returnn/config.py

```diff
@@ -131,6 +131,58 @@ def get(self, key, default=None):
             return self.post_config[key]
         return self.config.get(key, default)
 
+    def update(self, other):
+        """
+        updates a ReturnnConfig with an other ReturnnConfig
+
+        :param ReturnnConfig other:
+        """
+        self.config.update(other.config)
+        self.post_config.update(other.post_config)
+        if other.staged_network_dict is not None:
+            self.staged_network_dict = other.staged_network_dict
+
+        if self.python_prolog is None:
+            self.python_prolog = other.python_prolog
+        elif isinstance(self.python_prolog, "str") and isinstance(
+            other.python_prolog, "str"
+        ):
+            self.python_prolog += other.python_prolog
+        else:
+            self.python_prolog = [self.python_prolog, other.python_prolog]
+        if self.python_prolog_hash is None:
+            self.python_prolog_hash = other.python_prolog_hash
+        if isinstance(self.python_prolog_hash, "str") and isinstance(
+            other.python_prolog_hash, "str"
+        ):
+            self.python_prolog_hash += other.python_prolog_hash
+        else:
+            self.python_prolog_hash = [
+                self.python_prolog_hash,
+                other.python_prolog_hash,
+            ]
+
+        if isinstance(self.python_epilog, "str") and isinstance(
+            other.python_epilog, "str"
+        ):
+            self.python_epilog += other.python_epilog
```

Comment:
same as `python_prolog`?

Reply:
As the default for epilog is `""` there will be the case here that either of them or both are empty. We would then insert unnecessary newlines.

I could extend this code further by checks if either is empty and then only take the other..